### PR TITLE
[CPU][ARM] Enable subgraph tests

### PR DESCRIPTION
**Pull request by alvoron**
### Details:
 - Subgraph tests are enabled on ARM platforms

### Tickets:
 - CVS-112267

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/merge_transpose_reorder.cpp [hunk 1]
```diff
@@ -152,8 +152,17 @@ std::vector<InputShape> static_shapes = {
     InputShape{{}, {{1, 32, 16, 16}}},
 };
 
+#if defined(OPENVINO_ARCH_ARM)
+const ExpectedResult successfull_fuse_result{1, 1, 3};
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/merge_transpose_reorder.cpp:156]: It seems the reason is an additional reorder appears in the graph because the Eltwise (bias?) is not fused into groupconvolution node

**Review by EgorDuplensky — COMMENTED** [review]
Can we try to enable causal mask preprocess fusion for ARM as well?
@usstq What do you think?

**Review by EgorDuplensky — COMMENTED** [review]
@maxnick Can we use a planar layout for ReLuConcatConvSumInPlaceTest  subgraph test? Since currently the test forces blocking layout it is impossible to run it on ARM

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/conv_concat.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2018-2024 Intel Corporation
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/conv_concat.cpp:1]: Why we have to make those instances x86 specific?

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/fuse_transpose_reorder.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    auto transpose = std::make_shared<ov::op::v1::Transpose>(add, constOrder);
+    transpose->get_rt_info() = makeCPUInfo({memFmt}, {memFmt}, {});
+
+    ov::ResultVector results{std::make_shared<ov::op::v0::Result>(transpose)};
+    function = std::make_shared<ov::Model>(results, params, "TransposeReorder");
+}
+
+TEST_P(FuseTransposeAndReorderTest3, CompareWithRefs) {
+    run();
+    check_transpose_count(1);
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/fuse_transpose_reorder.cpp:69]: what result do we get for ARM?

**Review by EgorDuplensky — COMMENTED** [review]
let's remove layout specialization from interaction subgraph test and enable it for ARM as well.

**Comment by maxnick** [comment]
> ReLuConcatConvSumInPlaceTest

Yes, sure, but please use **nspc** layout instead as the main one for optimized convolution implementations.

**Comment by alvoron** [comment]
> let's remove layout specialization from interaction subgraph test and enable it for ARM as well.

As discussed offline, Interaction node and interaction transformations are enabled on x64 only, so Interaction tests are x64 specific as well.

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/fuse_transpose_reorder.cpp (hunk 3 above)
Comment (alvoron) [reply src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/fuse_transpose_reorder.cpp:69]: There is no Transpose on ARM.

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/conv_concat.cpp (hunk 2 above)
Comment (alvoron) [reply src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/x64/conv_concat.cpp:1]: Reference implementation (`ref_any`) does not support nchw/ndchw layouts on ARM. I've got the error `No supported primitive descriptors matched the provided input / output memory format filters` while pushing these layouts into the test.
nspc works well for reference on both ARM and x64. Can we replace nchw/ndchw with nhwc/ndhwc and keep conv tests common?

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/merge_transpose_reorder.cpp (hunk 1 above)
Comment (alvoron) [reply src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/merge_transpose_reorder.cpp:156]: Yes, `EltwiseAdd ` is not in the list of supported postops: https://github.com/alvoron/openvino/blob/37ab2c6abe2519e1ae2494e163798dfce7d8fce4/src/plugins/intel_cpu/src/dnnl_extension_utils.cpp#L227-L234

File: src/plugins/intel_cpu/tests/functional/utils/x64/filter_cpu_info.cpp [hunk 4]
```diff
@@ -81,4 +81,27 @@ std::vector<CPUSpecificParams> filterCPUInfoForDeviceWithFP16(const std::vector<
...
     return specificParams;
 }
+
+std::vector<CPUSpecificParams> filterCPUSpecificParams(const std::vector<CPUSpecificParams>& paramsVector) {
+    std::vector<CPUSpecificParams> filteredParamsVector = paramsVector;
+
+    if (!ov::with_cpu_x86_avx512f()) {
+        static const std::vector<cpu_memory_format_t> non_supported_f = {cpu_memory_format_t::nCw16c,
+                                                                         cpu_memory_format_t::nChw16c,
+                                                                         cpu_memory_format_t::nCdhw16c};
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/tests/functional/utils/x64/filter_cpu_info.cpp:91]: Could you please leave a comment regarding the reasoning

File: src/plugins/intel_cpu/tests/functional/utils/cpu_test_utils.cpp [hunk 5]
```diff
@@ -410,30 +410,6 @@ void CPUTestsBase::updateSelectedType(const std::string& primitiveType,
     selectedType += execType.get_type_name();
 }
 
-std::vector<CPUSpecificParams> filterCPUSpecificParams(const std::vector<CPUSpecificParams>& paramsVector) {
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/tests/functional/utils/cpu_test_utils.cpp:413]: Let's revert this change and fix the particular test cases which cause the issue

Comment (alvoron) [reply src/plugins/intel_cpu/tests/functional/utils/cpu_test_utils.cpp:413]: This change is reverted (this method has been moved to `utils/x64/filter_cpu_info.cpp`
2 methods - `getCPUParamsBlocked4D` and `getCPUParamsBlocked5D` were implemented in `single_layer_tests/slice.cpp` to do x64-specific fintering

**Review by EgorDuplensky — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]
